README.adoc

```diff
@@ -36,30 +38,238 @@ include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/
 ----
 
-NOTE: The above example does not specify `GET` vs. `PUT`, `POST`, and so forth, because `@RequestMapping` maps all HTTP operations by default. Use `@RequestMapping(method=GET)` to narrow this mapping.
+After the channel adapter is instantiated, an output channel where the adapter sends the received
+messages to must be configured. This channel can be explicitly declared through a bean.
 
+`src/main/java/hello/PubSubApplication.java`
+[source,java]
+----
+include::/complete/src/main/java/hello/PubSubApplication.java[tag=pubsubInputChannel]
+----
 
-== Make the application executable
+The `"testSubscription"` subscription must be created before the application is run. New
+subscriptions can be created through the
+https://console.cloud.google.com/cloudpubsub/subscriptions[Google Cloud Console] or,
+programatically, with the `PubSubAdmin` class.
 
-Although it is possible to package this service as a traditional link:/understanding/WAR[WAR] file for deployment to an external application server, the simpler approach demonstrated below creates a standalone application. You package everything in a single, executable JAR file, driven by a good old Java `main()` method. Along the way, you use Spring's support for embedding the link:/understanding/Tomcat[Tomcat] servlet container as the HTTP runtime, instead of deploying to an external instance.
+The message acknowledgement mode is set in the adapter to automatic, by default. The example shows
+the acknowledgement mode set to manual just to show how to manually acknowledge messages since, in
+the automatic mode, nothing needs to be done.
 
+Attached to an inbound channel is a service activator which is used to process incoming messages.
 
-`src/main/java/hello/Application.java`
+`src/main/java/hello/PubSubApplication.java`
 [source,java]
 ----
-include::complete/src/main/java/hello/Application.java[]
+include::/complete/src/main/java/hello/PubSubApplication.java[tag=messageReceiver]
 ----
 
+The `ServiceActivator` input channel name must match the method name of the inbound channel declared
+above. In this example, the message body is simply logged and the message is acknowledged. The
+acknowledgement is done through the `AckReplyConsumer` object, which is sent in the message headers.
+
+== Create an outbound channel adapter
+
+An outbound channel adapter listens to new messages in an application's Spring channel and publishes
+them to a Google Cloud Pub/Sub topic.
+
+Instantiating an outbound channel adapter requires a `PubSubTemplate`, which is Spring's abstraction
+to publish messages to Google Cloud Pub/Sub topics. The Spring Cloud GCP Pub/Sub Boot starter
+provides a pre-configured `PubSubTemplate`.
+
+`src/main/java/hello/PubSubApplication.java`
+[source,java]
+----
+include::/complete/src/main/java/hello/PubSubApplication.java[tag=messageSender]
+----
+
+After the outbound channel adapter is instantiated, the name of the Google Cloud Pub/Sub topic
```

Comment:
As mentioned before, lets cover topic creation, subscription creation at the top. A subscription requires a topic and so sequencing is weird when you get here.